packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/simulationservice/interventions/Intervention.java

```diff
@@ -32,4 +35,43 @@ public Intervention clone() {
 		}
 		return intervention;
 	}
+
+	// Check for no negative timesteps
+	// Check for no duplicate time + appliedTo pairs in static interventions
+	public Boolean validateIntervention() throws Exception {
+		//Sort static list
+		Collections.sort(this.staticInterventions, (staticOne, staticTwo) -> {
+			final String stringOne = staticOne.getAppliedTo() + staticOne.getTimestep().toString();
+			final String stringTwo = staticTwo.getAppliedTo() + staticTwo.getTimestep().toString();
+			return stringOne.compareTo(stringTwo);
+		});
```

Comment:
What is the desired sorting order here? Cause you are sorting strings with a numerical component and it rarely does what you want.

Reply:
Did not mean to push these changes yet! still playing with this idea

The idea being that if we sort first we can compare `i` and `i - 1` rather than `i` and `j`
Sorting here would mean that we put any duplicate pair beside eachother 
Duplicate pair means appliedTo and timestep are identical

Reply:
I would just use Set or Map for duplicate checks, then it is order-invariant and you save yourself some computation-cycles as well. You are saving a bit of resources in terms of space-complexity doing it your way, but it is largely negligible in the grand scheme of things and it increases computational complexity.